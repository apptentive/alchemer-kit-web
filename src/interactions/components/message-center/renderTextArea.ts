/**
 * Renders the Message Center textarea.
 *
 * @param {string} placeholderText - The text to use for the placeholder
 * @returns {HTMLDivElement} - The textarea container DOM Node.
 */
const renderTextArea = (placeholderText: string) => {
  const textarea = document.createElement('textarea');
  textarea.classList.add('apptentive-textinput');
  textarea.name = 'response';
  textarea.required = true;
  textarea.placeholder = placeholderText;
  textarea.rows = 4;
  textarea.addEventListener('blur', (event) => {
    (event.target as HTMLInputElement).classList.remove('apptentive-textinput--error');
  });

  const container = document.createElement('div');
  container.classList.add('apptentive-message-center-feedback');
  container.append(textarea);

  return container;
};

export default renderTextArea;
