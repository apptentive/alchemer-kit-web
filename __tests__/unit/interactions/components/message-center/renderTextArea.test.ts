import renderTextArea from '../../../../../src/interactions/components/message-center/renderTextArea';

const _placeholderText = 'Hint text';

describe('renderTextArea', () => {
  test('properly renders textarea', () => {
    const output = renderTextArea(_placeholderText);

    expect(output).toMatchSnapshot();
  });

  test('properly attaches blue event', () => {
    document.body.append(renderTextArea(_placeholderText));

    const textarea = document.querySelector<HTMLTextAreaElement>('.apptentive-textinput')!;
    textarea.classList.add('apptentive-textinput--error');
    textarea.focus();
    textarea.blur();

    expect(textarea.classList.contains('apptentive-textinput--error')).toBe(false);

    document.body.innerHTML = '';
  });
});
