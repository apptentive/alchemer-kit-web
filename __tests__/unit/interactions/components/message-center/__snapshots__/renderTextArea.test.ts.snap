// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderTextArea properly renders textarea 1`] = `
<div
  class="apptentive-message-center-feedback"
>
  <textarea
    class="apptentive-textinput"
    name="response"
    placeholder="Hint text"
    required=""
    rows="4"
  />
</div>
`;
